helm/cluster-api-provider-aws/templates/crd-install/crd-ciliumnetworkpolicy.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.ciliumNetworkPolicy.enabled }}
+apiVersion: cilium.io/v2
+kind: CiliumNetworkPolicy
+metadata:
+  name: {{ include "capa.crdInstall" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    # create hook dependencies in the right order
+    "helm.sh/hook-weight": "-2"
+    {{- include "capa.CRDInstallAnnotations" . | nindent 4 }}
+  labels:
+    app.kubernetes.io/component: {{ include "capa.crdInstall" . | quote }}
+    {{- include "capa.selectorLabels" . | nindent 4 }}
+    role: {{ include "capa.CRDInstallSelector" . | quote }}
+spec:
+  endpointSelector:
+    matchLabels:
+      app.kubernetes.io/component: {{ include "capa.crdInstall" . | quote }}
+      {{- include "capa.selectorLabels" . | nindent 6 }}
+  egress:
+    - toEntities:
+        - kube-apiserver
```

Comment:
I thought we needed a regular `NetworkPolicy` to access the api server for the CRD job.

Reply:
Why? Do you have a source where we had issues?

Reply:
I remember we have problems with network policies that used the `ipBlock` configuration, as that was not supported on cilium network policies, and we used `ipBlock` when allowing access to the api server. [This is the only thing](https://github.com/giantswarm/roadmap/issues/2201) I could find. Maybe @whites11 or @paurosello have more knowledge on this.

Reply:
so the problem here with k8s network policies, if you use ipBlock it only works for non-cluster IPs.
I'd say if you  tested this policy and it works we're good, otherwise you might need to whitelist egress for this app